ledger/eval/eval.go

```diff
@@ -1544,15 +1578,23 @@ func (validator *evalTxValidator) run() {
 // Validate: Eval(ctx, l, blk, true, txcache, executionPool)
 // AddBlock: Eval(context.Background(), l, blk, false, txcache, nil)
 // tracker:  Eval(context.Background(), l, blk, false, txcache, nil)
-func Eval(ctx context.Context, l LedgerForEvaluator, blk bookkeeping.Block, validate bool, txcache verify.VerifiedTransactionCache, executionPool execpool.BacklogPool) (ledgercore.StateDelta, error) {
+func Eval(ctx context.Context, l LedgerForEvaluator, blk bookkeeping.Block, validate bool, txcache verify.VerifiedTransactionCache, executionPool execpool.BacklogPool, cfg config.Local) (ledgercore.StateDelta, error) {
 	// flush the pending writes in the cache to make everything read so far available during eval
 	l.FlushCaches()
+	var granularEval = false
+	var txnDeltaTracer logic.EvalTracer
+	if cfg.EnableGranularEval {
+		granularEval = true
+		txnDeltaTracer = TxnGroupDeltaTracerForConfig(cfg)
+	}
```

Comment:
Being able to pass all the functionality in this way is really cool. Presumably we'll need access to this object in the API eventually, maybe `Eval` should accept the tracer instead? Would `GranularEval` be enabled implicitly if a tracer is used, or maybe the tracer could have a function to request it?